From ae0f8ba5d1c620ccad8ee9fbf9713ce13ec91c66 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Thu, 18 Mar 2021 17:04:28 +0100
Subject: [PATCH] Audiomanager: adjust music folder paths for LibreELEC-RR

---
 es-core/src/AudioManager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/es-core/src/AudioManager.cpp b/es-core/src/AudioManager.cpp
index dd1273456..cc2f97666 100644
--- a/es-core/src/AudioManager.cpp
+++ b/es-core/src/AudioManager.cpp
@@ -150,13 +150,13 @@ void AudioManager::playRandomMusic(bool continueIfPlaying)
 
 	std::vector<std::string> musics;
 	
-	// check in RetroPie music directory
+	// check in storage music directory
 	if(musics.empty())
-		getMusicIn(Utils::FileSystem::getHomePath() + "/RetroPie/roms/music", musics);
+		getMusicIn("/storage/music/emulationstation", musics);
   
 	// check in system sound directory
 	if(musics.empty())
-		getMusicIn("/usr/share/RetroPie/music", musics);
+		getMusicIn("/usr/share/emulationstation/music", musics);
   
 	// check in .emulationstation/music directory
 	if(musics.empty())
