--- /engine/source/utils.c
+++ /engine/source/utils.c
@@ -304,7 +304,7 @@
         writeToLogFile("Out of memory!\n");
         writeToLogFile("Allocation of size %i failed in function '%s' at %s:%i.\n", size, func, file, line);
 #ifndef WIN
-        writeToLogFile("Memory usage at exit: %u\n", mallinfo().arena);
+        writeToLogFile("Memory usage at exit: %u\n", mallinfo2().arena);
 #endif
         borExit(2);
     }

