From 36f41cf4cb51d7659df3a3aaea9721682402a061 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Wed, 22 Dec 2021 14:08:13 +0100
Subject: [PATCH] libretro/Makefile: only set LD, AR, RANLIB if not already set

---
 backends/platform/libretro/build/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/backends/platform/libretro/build/Makefile b/backends/platform/libretro/build/Makefile
index 25d141eb551e..5ab4d0a3f452 100644
--- a/backends/platform/libretro/build/Makefile
+++ b/backends/platform/libretro/build/Makefile
@@ -66,9 +66,9 @@ else
 endif
 endif
 
-LD        = $(CXX)
-AR        = ar cru
-RANLIB    = ranlib
+LD        ?= $(CXX)
+AR        ?= ar cru
+RANLIB    ?= ranlib
 
 ifeq ($(platform), unix)
    TARGET  := $(TARGET_NAME)_libretro.so
