--- a/backends/platform/libretro/build/Makefile
+++ b/backends/platform/libretro/build/Makefile
@@ -270,7 +270,8 @@
    HAVE_MT32EMU = 0
 else ifneq (,$(findstring armv,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
-   DEFINES += -fPIC -Wno-multichar -D_ARM_ASSEM_
+   CXXFLAGS := -std=c++11
+   DEFINES += -fPIC -Wno-multichar -D_ARM_ASSEM_ -DUSE_CXX11
    LDFLAGS += -shared -Wl,--version-script=../link.T -fPIC
    USE_VORBIS = 0
    USE_THEORADEC = 0

