From 1dd30b73cae1e54808209bbb55b511532a96bcf1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Wed, 31 Jan 2018 12:49:21 +0100
Subject: [PATCH] retroarch: Patch to fix keyboard typing

diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index bcdacb60a6..d2f00114bd 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -76,7 +76,7 @@
 #include "../../retroarch.h"
 #include "../../verbosity.h"
 
-#if defined(HAVE_XKBCOMMON) && defined(HAVE_KMS)
+#if defined(HAVE_XKBCOMMON)
 #define UDEV_XKB_HANDLING
 #endif
 
@@ -1402,7 +1402,7 @@
    /* Force xkb_handling on Lakka */
    udev->xkb_handling = true;
 #else
-   udev->xkb_handling = string_is_equal(ctx_ident.ident, "kms");
+   udev->xkb_handling = true;
 #endif /* HAVE_LAKKA */
 #endif
 

