From 714f0706ba1f76a462b3fa116628bc0124c978e3 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Wed, 1 Dec 2021 13:52:53 +0100
Subject: [PATCH] CMAkeLists.txt: use "prefix" path for pkgconfig/sdl2-config

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9411c76e79..73d333fce8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2483,9 +2483,9 @@ if(NOT WINDOWS OR CYGWIN OR MINGW)
   set(prefix ${CMAKE_INSTALL_PREFIX})
 
   set(exec_prefix "\${prefix}")
-  set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
+  set(libdir "\${prefix}/lib")
   set(bindir "${CMAKE_INSTALL_FULL_BINDIR}")
-  set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
+  set(includedir "\${prefix}/include")
   if(SDL_STATIC)
     set(ENABLE_STATIC_TRUE "")
     set(ENABLE_STATIC_FALSE "#")
