--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,6 @@
 MANDIR = $(PREFIX)/share/man/man1
 LICENSEDIR = $(PREFIX)/share/licenses/$(TARGET)
 
-CC = gcc
 LD = $(CC)
 PKG_CONFIG = pkg-config
 
@@ -33,7 +32,7 @@
 .NOTPARALLEL:
 
 .PHONY: all
-all: clean $(TARGET) mans
+all: clean $(TARGET)
 
 .PHONY: $(TARGET)
 $(TARGET): $(OBJS)
@@ -43,10 +42,8 @@
 	$(CC) $(CPPFLAGS) $(CFLAGS) -o "$@" -c "$<"
 
 .PHONY: install
-install: $(TARGET) mans
+install: $(TARGET)
 	$(INSTALL) -Dm 0755 "$(TARGET)" "$(DESTDIR)$(BINDIR)/$(TARGET)"
-	$(INSTALL) -Dm 0644 man/unclutter-xfixes.1 "$(DESTDIR)$(MANDIR)/unclutter.1"
-	$(INSTALL) -Dm 0644 -t "$(DESTDIR)$(LICENSEDIR)/" LICENSE
 
 .PHONY: uninstall
 uninstall:

